src/errors.rs

```diff
@@ -0,0 +1,8 @@
+//! All possible errors VAD might encounter.
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum VadError {
+    #[error("Float sample must be in the range -1.0 to 1.0")]
+    FloatSampleNotInRangeMinus1To1,
```

Comment:
I would give this a simpler type name like `InvalidData` or `SampleOutOfRange` and elaborate in the error string. As the silero model updates we can't guarantee the range will stay the same and such a rigid name would result in unavoidable semver breakage